### ISSUE TITLE
Consider switching from clirr to something else, like japicmp

### ISSUE THREAD

**Issue by Flowdalic** [aalmiray/clirr-gradle-plugin#9]
I ♥ clirr-gradle-plugin, especially since [it implemented the nice heuristic I asked for](https://github.com/aalmiray/clirr-gradle-plugin/issues/3#issuecomment-76953925), but [clirr](http://clirr.sourceforge.net) doesn't seem to be actively developed any more.

How about switching to an alternative? I did some research, and so far, candidates, with recent commit activity, are

- https://github.com/siom79/japicmp
- https://lvc.github.io/japi-compliance-checker/

On a first look, japicmp looks promising.

**Comment by aalmiray** [comment]
Yes, Clirr is pretty much unsupported software. I even had to patch it's BCEL dependency with a fork from Google instead of the official Apache because version 6.0 was not released at the time. It looks like version 6.1 is finally out so I may need to repackage the plugin once more.

AFAIK `japicmp` works at the source level, just like jdiff does. Clirr works at the bytecode level.
`japi-compliance-checker` looks to be a standalone binary that targets specific platforms. It wojld be a nightmare to setup on CI.

**Comment by Flowdalic** [comment]
> AFAIK japicmp works at the source level, just like jdiff does.

Not sure, it seems to take two jar files as input.

**Comment by aalmiray** [comment]
I see it now, my mistake, I thought `japicmp` was source based for some reason. It turns out @melix wrote a gradle plugin for it https://github.com/melix/japicmp-gradle-plugin
